quantumlib/Cirq
# Freshen-up cirq dependencies

- allow latest scipy because quimb>=1.18 works with scipy-1.13
  (context in #6543)
- allow latest rstcheck
- fix dependency specification for tensorflow-docs so it works with
  `uv pip compile`
- Enable the current latest versions of notebook, ipykernel, papermill
  which are only used for notebook tests
- Add temporary specifications for notebook-related transitive dependencies
  to help pip dependency resolver (should be redundant after #6281) 

Partially fixes #6464
